apps/server/src/modules/tool/tool-launch/service/tool-launch.service.ts

```diff
@@ -83,17 +87,37 @@ export class ToolLaunchService {
 		};
 	}
 
-	private isToolStatusLatestOrThrow(
+	private async determineToolConfigurationStatusThroughValidation(
+		schoolExternalTool: SchoolExternalTool,
+		contextExternalTool: ContextExternalTool
+	): Promise<ToolConfigurationStatus> {
+		try {
+			await this.schoolExternalToolValidationService.validate(schoolExternalTool);
+			await this.contextExternalToolValidationService.validate(contextExternalTool);
+			return ToolConfigurationStatus.LATEST;
+		} catch (err) {
+			return ToolConfigurationStatus.OUTDATED;
+		}
+	}
+
+	private async isToolStatusLatestOrThrow(
 		userId: EntityId,
 		externalTool: ExternalTool,
 		schoolExternalTool: SchoolExternalTool,
 		contextExternalTool: ContextExternalTool
-	): void {
-		const status: ToolConfigurationStatus = this.commonToolService.determineToolConfigurationStatus(
-			externalTool,
-			schoolExternalTool,
-			contextExternalTool
-		);
+	): Promise<void> {
+		let status: ToolConfigurationStatus;
+
+		if (Configuration.get('FEATURE_COMPUTE_TOOL_STATUS_WITHOUT_VERSIONS_ENABLED')) {
```

Comment:
Please add a comment and link ticket N21-1337 for deletion of this if-statement, the feature flag and the version property at the entity. Extend the ticket if something of these is missing